### DISCUSSION TITLE
Music assistant and ledfx

### DISCUSSION THREAD

**Discussion by seancmalloy** [music-assistant/support#1496] [Q&A]
Does anyone know the best way to pass audio from music assistant to ledfx running on remote server (debian with pulse)? Unfortunately ledfx is no longer available as add-on so I need to send it over local network.  As anyone had to set this up after ledfx add-on got benched?

Update - 

Decided to go different route. Installed shairport-sync with pulse and added appletv media player to ha. Works good with music assistant but throws up - Failed to call service media_player/play_media. HTTP/1.1 method POST failed with code 400: Bad Request when trying to use ha integrated media. does anyone know why?

service: media_player.media_play
shairport-sync device is not an option

service: media_player.media_play_pause
data: {}
target:
  device_id: c2bb63da1150ef6e347e284fa5d8513e
  entity_id: media_player.orangepizero2

Failed to call service media_player.media_play_pause. Entity media_player.orangepizero2 does not support this service.

**Comment by OzGav** [comment]
Did you try the play service call vice play_pause?

**Reply by marcelveldt** [reply]
Using the apple tv media player in HA for playback is very unstable and I really would not recommend it, mostly because its buffering what  is supposed to be streaming.

Instead you can use the Airplay player directly in MA

**Reply by marcelveldt** [reply]
Or make it less complicated by just using squeezelite ofcourse

**Comment by seancmalloy** [comment]
yes, you may be correct. although i first tried with no success with squeezelite. i used the debain pulseaudio-squeezelite pkg but ma didn't find it. shairport-sync seems to work ok and audio quality doesn't really matter since it is just being dumped into a null sink for ledfx. i still need to setup a few led strips to test and determine if the buffering throws off the timing/sync. if the sync is off ill try squeezelite again

and as far as shairport-sync and the integrated media in ha, yes i tried to test the compatible functionality between the two. when i go to dev tools > services > and try to call media_play - shairport-sync is NOT a device option. but with media_play_pause shairport-sync IS a device option and completes

update - new error
now when i try to use integrated media with shairport-sync i get - Failed to call service media_player/play_media. not connected to remote. 

thnx

**Comment by Slim1399** [comment]
Hi,
I think LedFx should one day have its place in Music Assistant; a plugin would be great !
